blaze/expr/expressions.py

```diff
@@ -790,6 +804,92 @@ def coerce(expr, to):
     return Coerce(expr, dshape(to) if isinstance(to, _strtypes) else to)
 
 
+class Cast(Expr):
+    """Cast an expression to a different type.
+
+    This is only an expression time operation.
+    """
+    __slots__ = '_hash', '_child', 'to'
+
+    def _dshape(self):
+        return self.to
+
+    def __str__(self):
+        return 'cast(%s, to=%r)' % (self._child, str(self.to))
+
+
+@copydoc(Cast)
+def cast(expr, to):
+    return Cast(expr, dshape(to) if isinstance(to, _strtypes) else to)
+
+
+Expr.cast = cast  # method of all exprs
+
+
+def binop_name(expr):
+    if not isscalar(expr.dshape.measure):
+        return None
+    l = getattr(expr.lhs, '_name', None)
+    r = getattr(expr.rhs, '_name', None)
+    if bool(l) ^ bool(r):
+        return l or r
+    elif l == r:
+        return l
+
+    return None
+
+
+def binop_inputs(expr):
+    if isinstance(expr.lhs, Expr):
+        yield expr.lhs
+    if isinstance(expr.rhs, Expr):
+        yield expr.rhs
+
+
+class Coalesce(Expr):
+    """SQL like coalesce.
+
+    coalesce(a, b) = {
+        a if a is not NULL
+        b otherwise
+    }
+    """
+    __slots__ = '_hash', 'lhs', 'rhs', 'dshape'
+    __inputs__ = 'lhs', 'rhs'
```

Comment:
This is variadic in sql but we don't yet have a good way to handle this in blaze. When we do support that this would be a good candidate for a variadic (and then concat)